### PR TITLE
Feature/api 353 swap to postgres

### PR DESCRIPTION
**Pull request by Pdzly**

### CODE REVIEW

**Comment by Pdzly** [comment]
I am not sure if test: ["CMD-SHELL", "pg_isready", "-d", "backend"] is correct.

**Comment by Pdzly** [comment]
I found it out, but i forgot that the federation uses the mysql db too, i will ask them if they are going to swap over postgresql too, or if they stay at mysql

**Comment by lazyguru** [comment]
We are. We are nearly agnostic to DB

**Review by jgrim — DISMISSED** [review]

File: common/services.yml [hunk 1]
```diff
@@ -11,17 +11,20 @@ x-logging: &default-logging
...
-    image: mysql:latest
+    image: postgres:latest
     environment:
-      - MYSQL_ROOT_PASSWORD=root
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=root
+      - PGUSER=root
+      - POSTGRES_DB=backend
     volumes:
       - ./db:/docker-entrypoint-initdb.d
```
Comment (lazyguru) [inline common/services.yml:19]: The file in here likely needs syntax updates. Did you test running things?

**Review by lazyguru — DISMISSED** [review]

File: common/services.yml (hunk 1 above)
Comment (lazyguru) [reply common/services.yml:19]: I update the script. I also removed some variables here as they are not needed

**Comment by lazyguru** [comment]
Federation PR: https://github.com/sublinks/sublinks-federation/pull/80

NOTE: I made changes to this PR so that it now uses schemas instead of databases. Not sure if this is the right way to do it in Postgres as I am much, much more comfortable in MySQL but this is how I seemed to have to do it since unlike MySQL, Postgres doesn't seem to support switching databases in a SQL script (or even applying permissions to a schema inside a different database from the one you are connected too).

**Review by jgrim — APPROVED** [review]

**Comment by lazyguru** [comment]
Federation PR is merged. @Pdzly you can merge this one as soon as the API side is ready

**Pull request merged**